src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstruct.cs

```diff
@@ -30,9 +30,27 @@ internal sealed class SynthesizedRecordDeconstruct : SynthesizedRecordOrdinaryMe
 
         protected override DeclarationModifiers MakeDeclarationModifiers(DeclarationModifiers allowedModifiers, BindingDiagnosticBag diagnostics)
         {
-            const DeclarationModifiers result = DeclarationModifiers.Public;
+            var result = DeclarationModifiers.Public;
+
+            if (ContainingType.IsRecordStruct && !_positionalMembers.Any(m => hasNonReadOnlyGetter(m)))
```

Comment:
Looks like `_positionalMembers` is not initialized here for some reason.

Reply:
Had to change the approach to accept `isReadOnly` parameter into the constructor and add a modifier outside of `MakeDeclarationModifiers` method.